### ISSUE TITLE
log: Choose logging framework and supporting utilities

### ISSUE THREAD

**Issue by Wondertan** [celestiaorg/celestia-node#39]
For obvious reasons, we have to choose a way of logging things. Instead of arguing on which library to use, let's first agree with the most convenient way of using a logger for us. Currently, I see only two options, but feel free to give others(NOTE: Each option would provide identical output.):
* Passing logger in each component/service constructor and keeping it in structures, like:
```go
package foo

// def
type FooService struct {
	...
	log log.Logger
}

func (fs *FooService) Foo() {
	fs.log.Debug("Foo fooed!")
}

// usage
func logic() {
	toplvlLogger := log.Logger("")
	fs := &FooService(..., toplvlLogger.With("foo"))
	fs.Foo()
}

```

* Defining logger as private per-package globar var, like:
```go
// def
package foo

var log = logging.Logger("foo")

type FooService struct {
	...
}

func (fs *FooService) Foo() {
	log.Debug("Foo fooed!")
}

// usage 
func logic() {
	fs := &FooService(...)
	fs.Foo()
}
```
 
Personally, the second way seems much cleaner to me and my favorite, as it does not require us to pass the logger, everywhere, all the time.

WDYT?

**Comment by renaynay** [comment]
Let's do option number 2, looks cleaner to me.

**Comment by renaynay** [comment]
Documenting some notes re: which logging dep we should use: 

https://github.com/celestiaorg/celestia-node/pull/29/files#r701006369

**Comment by liamsi** [comment]
referencing my comment here: https://github.com/celestiaorg/celestia-node/pull/51#discussion_r707315717

tldr: even though 2 looks cleaner, let's not use global vars (as they might have side effects, also in tests etc) and let's use the tendermint logger unless there are very good reasons for an alternative. Let's discuss about loggers after the devnet IMO.

**Comment by Wondertan** [comment]
As per sync discussion we decided not to change anything and keep things as they are in the master.
